sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/ProjectTypeHelper.cs

```diff
@@ -54,6 +54,7 @@ internal static bool IsTestAssemblyName(string assemblyName)
         {
             "MICROSOFT.VISUALSTUDIO.TESTPLATFORM.TESTFRAMEWORK",
             "MICROSOFT.VISUALSTUDIO.QUALITYTOOLS.UNITTESTFRAMEWORK",
+            "XUNIT", // Legacy Xunit (v1.x)

```

Comment:
This method is the classifier used to work out whether a project is a test project or not. Legacy XUnit projects were not being considered as test projects so test rules were not being run against them.